commit 2f1fd70a4e9ec65dc83a4d98dba1b3fc2be9c6ce
Author: Juan RP <xtraeme@gmail.com>
Date:   Fri Jun 21 17:13:06 2019 +0200

    xbps_transaction_prepare: initialize trans dict!
    
    ... otherwise it will always return ENXIO

diff --git a/lib/transaction_prepare.c b/lib/transaction_prepare.c
index 9921127d..79a68f2a 100644
--- lib/transaction_prepare.c
+++ lib/transaction_prepare.c
@@ -279,6 +279,9 @@ xbps_transaction_prepare(struct xbps_handle *xhp)
 	unsigned int i, cnt;
 	int rv = 0;
 
+	if ((rv = xbps_transaction_init(xhp)) != 0)
+		return rv;
+
 	if (xhp->transd == NULL)
 		return ENXIO;
 
